--reinstall-packages-from fails: "Could not install from "" as it does not contain a package.json file"

<!-- Thank you for being interested in nvm! Please help us by filling out the following form if you‘re having trouble. If you have a feature request, or some other question, please feel free to clear out the form. Thanks! -->

- Operating system and version: Fedora KDE 27

- `nvm debug` output:
<details>
<!-- do not delete the following blank line -->

```sh
nvm --version: v0.33.8
$SHELL: /bin/bash
$HOME: /home/[my name]
$NVM_DIR: '$HOME/.nvm'
$PREFIX: ''
$NPM_CONFIG_PREFIX: ''
$NVM_NODEJS_ORG_MIRROR: ''
$NVM_IOJS_ORG_MIRROR: ''
shell version: 'GNU bash, version 4.4.19(1)-release (x86_64-redhat-linux-gnu)'
uname -a: 'Linux 4.14.18-300.fc27.x86_64 #1 SMP Thu Feb 8 01:35:31 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux'
OS version: 
curl: /usr/bin/curl, curl 7.55.1 (x86_64-redhat-linux-gnu) libcurl/7.55.1 OpenSSL/1.1.0g zlib/1.2.11 libidn2/2.0.4 libpsl/0.18.0 (+libidn2/2.0.3) libssh2/1.8.0 nghttp2/1.25.0
wget: /usr/bin/wget, GNU Wget 1.19.4 built on linux-gnu.
git: /usr/bin/git, git version 2.14.3
grep: alias grep='grep --color=auto'
        /usr/bin/grep (grep --color=auto), grep (GNU grep) 3.1
awk: /usr/bin/awk, GNU Awk 4.1.4, API: 1.1 (GNU MPFR 3.1.5, GNU MP 6.1.2)
sed: /usr/bin/sed, sed (GNU sed) 4.4
cut: /usr/bin/cut, cut (GNU coreutils) 8.27
basename: /usr/bin/basename, basename (GNU coreutils) 8.27
rm: /usr/bin/rm, rm (GNU coreutils) 8.27
mkdir: /usr/bin/mkdir, mkdir (GNU coreutils) 8.27
xargs: /usr/bin/xargs, xargs (GNU findutils) 4.6.0
nvm current: v9.6.1
which node: $NVM_DIR/versions/node/v9.6.1/bin/node
which iojs: which: no iojs in (/opt/OpenPrinting-Gutenprint/sbin:/opt/OpenPrinting-Gutenprint/bin:$NVM_DIR/versions/node/v9.6.1/bin:/usr/lib64/qt-3.3/bin:/opt/OpenPrinting-Gutenprint/sbin:/opt/OpenPrinting-Gutenprint/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:$HOME/.local/bin:$HOME/bin)
which npm: $NVM_DIR/versions/node/v9.6.1/bin/npm
npm config get prefix: $NVM_DIR/versions/node/v9.6.1
npm root -g: $NVM_DIR/versions/node/v9.6.1/lib/node_modules
```
</details>

- `nvm ls` output:
<details>
<!-- do not delete the following blank line -->

```sh
         v9.6.0
->       v9.6.1
default -> v9.6.0
node -> stable (-> v9.6.1) (default)
stable -> 9.6 (-> v9.6.1) (default)
iojs -> N/A (default)
lts/* -> lts/carbon (-> N/A)
lts/argon -> v4.8.7 (-> N/A)
lts/boron -> v6.13.0 (-> N/A)
lts/carbon -> v8.9.4 (-> N/A)
```
</details>

- How did you install `nvm`? (e.g. install script in readme, homebrew): Install script I believe.  It's been a while.

- What steps did you perform?
1. `nvm ls-remote`
2. See there is a new version, e.g. v9.6.1
3. `nvm install v9.6.1 --reinstall-packages-from=v9.6.0`

- What happened?
```sh
Downloading and installing node v9.6.1...
Downloading https://nodejs.org/dist/v9.6.1/node-v9.6.1-linux-x64.tar.xz...
######################################################################## 100.0%
Computing checksum with sha256sum
Checksums matched!
Now using node v9.6.1 (npm v5.6.0)
Reinstalling global packages from v9.6.0...
npm ERR! code ENOLOCAL
npm ERR! Could not install from "" as it does not contain a package.json file.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/[my name]/.npm/_logs/2018-02-23T15_26_22_862Z-debug.log
Linking global packages from v9.6.0...
```

- What did you expect to happen?
Just install any previous packages like normal.  This used to work fine on Ubuntu but I am on Fedora now, not sure if that makes a difference.

- Is there anything in any of your profile files (`.bashrc`, `.bash_profile`, `.zshrc`, etc) that modifies the `PATH`?
No.

Just to confirm, is this output the raw or the modified? I wonder about the `""` here:
`npm ERR! Could not install from "" as it does not contain a package.json file.`

--- a/nvm.sh
+++ b/nvm.sh
@@ -3339,19 +3339,27 @@ nvm() {
       LINKS="${NPMLIST##* //// }"
 
       nvm_echo "Reinstalling global packages from $VERSION..."
-      nvm_echo "$INSTALLS" | command xargs npm install -g --quiet
+      if [ -n "${INSTALLS}" ]; then
+        nvm_echo "$INSTALLS" | command xargs npm install -g --quiet
+      else
+        nvm_echo "No installed global packages found..."
+      fi
 
       nvm_echo "Linking global packages from $VERSION..."
-      (
-        set -f; IFS='
+      if [ -n "${LINKS}" ]; then
+        (
+          set -f; IFS='
 ' # necessary to turn off variable expansion except for newlines
-        for LINK in $LINKS; do
-          set +f; unset IFS # restore variable expansion
-          if [ -n "$LINK" ]; then
-            (nvm_cd "$LINK" && npm link)
-          fi
-        done
-      )
+          for LINK in $LINKS; do
+            set +f; unset IFS # restore variable expansion
+            if [ -n "$LINK" ]; then
+              (nvm_cd "$LINK" && npm link)
+            fi
+          done
+        )
+      else
+        nvm_echo "No linked global packages found..."
+      fi
     ;;
     "clear-cache" )
       command rm -f "$NVM_DIR/v*" "$(nvm_version_dir)" 2>/dev/null

--- a/test/slow/nvm reinstall-packages/setup_dir
+++ b/test/slow/nvm reinstall-packages/setup_dir
@@ -3,6 +3,8 @@
 \. ../../../nvm.sh
 nvm install 0.10.28
 nvm install 0.10.29
+nvm install 4.7.1
+nvm install 4.7.2
 
 if [ -f ".nvmrc" ]; then
   mv .nvmrc .nvmrc.bak

--- a/test/slow/nvm reinstall-packages/teardown_dir
+++ b/test/slow/nvm reinstall-packages/teardown_dir
@@ -3,6 +3,8 @@
 \. ../../../nvm.sh
 nvm uninstall v0.10.28
 nvm uninstall v0.10.29
+nvm uninstall v4.7.1
+nvm uninstall v4.7.2
 
 rm -f .nvmrc
 

--- a/test/slow/nvm reinstall-packages/works with no installs
+++ b/test/slow/nvm reinstall-packages/works with no installs
@@ -1,0 +1,18 @@
+#!/bin/sh
+
+die () { echo "$@" ; exit 1; }
+
+\. ../../../nvm.sh
+
+get_packages() {
+  npm list -g --depth=0 | \sed -e '1 d' -e 's/^.* \(.*\)@.*/\1/' -e '/^npm$/ d' | xargs
+}
+
+nvm use 4.7.2
+ORIGINAL_PACKAGES=$(get_packages)
+
+nvm reinstall-packages 4.7.1
+FINAL_PACKAGES=$(get_packages)
+
+[ -z "${ORIGINAL_PACKAGES}" ] || die "original packages were not empty: ${ORIGINAL_PACKAGES}"
+[ -z "${FINAL_PACKAGES}" ] || die "final packages were not empty: ${FINAL_PACKAGES}"

